ko/lessons/advanced/umbrella-projects.md

```diff
@@ -0,0 +1,219 @@
+
+```shell
+$ mix new utilities --sup
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/utilities.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/utilities_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd utilities
+    mix test
+
+Run "mix help" for more commands.
+
+
+$ mix new datasets
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/datasets.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/datasets_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd datasets
+    mix test
+
+Run "mix help" for more commands.
+
+$ mix new svm --sup
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/svm.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/svm_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd svm
+    mix test
+
+Run "mix help" for more commands.
+```
+
+이제 아래와 같은 프로젝트 트리 구조를 갖게 됩니다.
+
+```shell
+$ tree
+.
+├── README.md
+├── apps
+│   ├── datasets
+│   │   ├── README.md
+│   │   ├── config
+│   │   │   └── config.exs
+│   │   ├── lib
+│   │   │   └── datasets.ex
+│   │   ├── mix.exs
+│   │   └── test
+│   │       ├── datasets_test.exs
+│   │       └── test_helper.exs
+│   ├── svm
+│   │   ├── README.md
+│   │   ├── config
+│   │   │   └── config.exs
+│   │   ├── lib
+│   │   │   └── svm.ex
+│   │   ├── mix.exs
+│   │   └── test
+│   │       ├── svm_test.exs
+│   │       └── test_helper.exs
+│   └── utilities
+│       ├── README.md
+│       ├── config
+│       │   └── config.exs
+│       ├── lib
+│       │   └── utilities.ex
+│       ├── mix.exs
+│       └── test
+│           ├── test_helper.exs
+│           └── utilities_test.exs
+├── config
+│   └── config.exs
+└── mix.exs
+```
+
+다시 엄브렐라 프로젝트 최상위 폴더로 이동해 보면, `compile` 과 같은 일반적인 Mix 명령을 계속해서 사용할 수 있다는 걸 알 수 있습니다. 엄브렐라 프로젝트 안에 있는 서브프로젝트는 전부 일반적인 애플리케이션이기 때문에, 프로젝트의 경로로 들어가서 Mix가 제공하는 똑같은 것들을 그대로 할 수 있습니다.
+
+```
+$ mix compile
+
+==> svm
+Compiled lib/svm.ex
+Generated svm app
+
+==> datasets
+Compiled lib/datasets.ex
+Generated datasets app
+
+==> utilities
+Compiled lib/utilities.ex
+Generated utilities app
+
+Consolidated List.Chars
+Consolidated Collectable
+Consolidated String.Chars
+Consolidated Enumerable
+Consolidated IEx.Info
+Consolidated Inspect
+```
+
+## IEx
+
+엄브렐라 프로젝트에서 앱과 상호작용을 하는 건 다른 프로젝트랑은 조금 다를 것 같다고 느끼실지도 모르겠습니다. 음 뭐랄까 그렇게 생각하셔도 나쁠 건 없지만, 사실은 그렇지 않을 수도 있다고요! 엄브렐라 프로젝트의 최상위 디렉터리로 올라가 `iex -S mix` 명령으로 IEx를 실행하면, 평범하게 프로젝트 안에 있는 전부와 상호작용할 수 있습니다. 한번 확인해보게 `apps/datasets/lib/datasets.ex` 파일을 수정해 봅시다.
```

Comment:
1) Eliminating `음 뭐랄까` can be good choice.
2) (Maybe) Rather than `프로젝트 안에 있는 전부와`, `모든 자식 프로젝트들과` is more proper. `프로젝트 안에 있는 전부와` seems ambiguous.